src/client.ts

```diff
@@ -281,14 +293,14 @@ export default class Client {
     const env = process.env;
     const useYjit = vscode.workspace.getConfiguration("rubyLsp").get("yjit");
 
-    // Enabling YJIT only provides a performance benefit on Ruby 3.2.0 and above
-    if (!useYjit || env.RUBY_VERSION! < "3.2.0") {
+    if (!this.ruby.rubyVersion) {
       return env;
     }
 
-    // Check if the Ruby was compiled with YJIT since it's not required. The exec must be sync here because this is used
-    // in the constructor, which cannot be an async function
-    if (!execSync("ruby -v --yjit").toString().includes("+YJIT")) {
+    const [major, minor, _patch] = this.ruby.rubyVersion.split(".");
+
+    // Enabling YJIT only provides a performance benefit on Ruby 3.2.0 and above
+    if (!useYjit || !this.ruby.yjitEnabled || !(major >= "3" && minor >= "2")) {
```

Comment:
I am afraid this is still not adequate since we are not comparing integers. `"10" > "2"` will still be `false`, for the minor comparison, for example.
```suggestion
    const [major, minor, _patch] = this.ruby.rubyVersion.split(".").map(Number);

    // Enabling YJIT only provides a performance benefit on Ruby 3.2.0 and above
    if (!useYjit || !this.ruby.yjitEnabled || (major == 3 && minor < 2)) {
```

Reply:
Or we can check it like `[major, minor] < [3, 2]` if you want.

Reply:
```js
> [2, 1] < [3, 2]
true
> [3, 1] < [3, 2]
true
> [3, 2] < [3, 2]
false
> [4, 1] < [3, 2]
false
```